src/encoder/mod.rs

```diff
@@ -620,10 +623,10 @@ impl<'a, W: 'a + Write + Seek, T: ColorType> ImageEncoder<'a, W, T> {
         height: u32,
     ) -> TiffResult<ImageEncoder<'a, W, T>> {
         let row_samples = u64::from(width) * u64::try_from(<T>::BITS_PER_SAMPLE.len())?;
-        let row_bytes = row_samples * u64::from(<T::Inner>::BYTE_LEN);
 
-        // As per tiff spec each strip should be about 8k long
-        let rows_per_strip = (8000 + row_bytes - 1) / row_bytes;
```

Comment:
I think it would be better to keep the default in terms of number of bytes, but just significantly increase the target. Maybe 256KB or 1MB or so?

Reply:
I checked multiple libraries and see the default for RowsPerStrip is the image height. What would be a good reason for using a custom arbitrary default?

Reply:
Indeed, for example ImageMagick uses that strategy. On the 'why not' though, it appears to have been the cause for [at least one denial-of-service weakness](https://github.com/ImageMagick/tiff/blob/19f25769da6bde2434e89305e7abbf2abf729996/html/v4.0.9.html#L221-L224). I'd rather have some limit here as well, and be it just to keep the expectation of handling multiple strips 'oiled' so to speak. Maybe the default limit we use for our decode buffer, that is 256 MB? That should hardly limit decoding throughput for quite some time while still being reasonably small in terms of total memory—even though *we* don't allocate the caller will and it's nice to nudge them in the right direction of considering OOM.

Reply:
For compressed images, you can only seek to the start of a strip. And decompressing large images can take a non-trivial amount of time so seeking to read only the part you need can be quite useful useful.

Reply:
Ok, then I'll set it to default 1MB strip size instead of the original 8KB. Agree?

Reply:
Sounds good to me.